spec/fixtures/generate_external_certs.sh

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash
+
+# This script re-generates all of the keys and certs within the external_certs directory.
+# The files there are used by the spec/features/settings/external_cert_feature_spec test
+
+function init() {
+  mkdir -p /tmp/ca_gen/csr
+  CSR_DIR=/tmp/ca_gen/csr
+  
+  mkdir -p /tmp/ca_gen/certs
+  DIR_CERTS=/tmp/ca_gen/certs
+  
+  mkdir -p /tmp/ca_gen/conf
+  DIR_CONF=/tmp/ca_gen/conf
+  
+  mkdir -p /tmp/ca_gen/keys
+  DIR_KEYS=/tmp/ca_gen/keys
+  
+  mkdir -p /tmp/ca_gen/cadata
+  DIR_CADATA=/tmp/ca_gen/cadata
+  
+  echo "dir cadata: $DIR_CADATA"
+}
+
+function cleanup() {
+  rm -rf /tmp/ca_gen
+}
+
+function init_ca() { # $1 ca_name
+  echo "Initiating CA  \"$1\""
+  generate_conf $1
+  # Create / clear certs directory for CA
+  mkdir -p $DIR_CADATA/$1/db.certs
+  rm $DIR_CADATA/$1/db.certs/* 2>/dev/null
+  
+  # Create / clear index file of certs
+  rm $DIR_CADATA/$1/db.inde* 2>/dev/null
+  touch $DIR_CADATA/$1/db.index
+  
+  # Other basic needed files
+  echo unique_subject = yes > $DIR_CADATA/$1/db.index.attr
```

Comment:
In some cases the db.index.attr gets automatically created. In others it does not. Unsure exactly why it doesn't get created in some cases so this is here to ensure it exists. This is the default content of the file when it does get automatically created.

Reply:
Added comment just that it is created with default content